internal/pkg/admission/reconciler.go

```diff
@@ -92,6 +92,13 @@ func (r *Reconciler) ReconcileDeletion(
 		errors = append(errors, err)
 	}
 
+	patch := policyServer.DeepCopy()
+	patch.ObjectMeta.Finalizers = nil
+	err = r.Client.Patch(ctx, patch, client.MergeFrom(policyServer))
```

Comment:
We should only remove the finalizer we know about. This allows for other controllers to react if they add finalizers too, for whatever reason.

I would add a constant with a `kubewarden` finalizer for example. On this logic, we get the list of finalizers and remove `kubewarden` when we are done. If we are the only ones, the list of finalizers will be empty, letting the Kubernetes GC remove the resource. Also, when we implement the mutating webhook, we can add the finalizer on the resource if not present.

Reply:
yes, great idea! done